test/HardETHVaultHandler.t.sol

```diff
@@ -360,4 +360,61 @@ contract ETHVaultHandlerTest is DSTest {
 		assertEq(debt, 0);
 		assertEq(owner, user);
 	}
+
+	function testSetBurnFee_ShouldRevert_WhenNotCalledByOwner() public {
+		//setUp
+		uint256 oldFee = ethVault.burnFee();
+		vm.expectRevert("Ownable: caller is not the owner");
+
+		//execution
+		ethVault.setBurnFee(2);
+
+		//assert
+		assertEq(oldFee, ethVault.burnFee());
+	}
+
+	function testSetBurnFee_ShouldRevert_WhenValueAboveMax(uint256 _newFee) public {
+		//setUp
+		if (_newFee <= 1000) {
+			return;
+		}
+		uint256 oldFee = ethVault.burnFee();
+		vm.expectRevert("VaultHandler::setBurnFee: burn fee higher than MAX_FEE");
+
+		//execution
+		orchestrator.setBurnFee(ethVault, _newFee);
+
+		//assert
+		assertEq(oldFee, ethVault.burnFee());
+	}
+
+	function testSetBurnFee_ShouldAllowDecimals_WhenValueBelowMax(uint256 _newFee) public {
+		//setUp
+		if (_newFee > 1000) {
+			return;
+		}
+
+		//execution
+		vm.expectEmit(true, true, true, true);
+		emit NewBurnFee(address(orchestrator), _newFee);
+		orchestrator.setBurnFee(ethVault, _newFee);
+
+		//assert
+		assertEq(ethVault.burnFee(), _newFee);
+	}
+
+	function testGetFee_ShouldCalculateCorrectValue(uint8 _burnFee, uint96 _amount) public {
```

Comment:
I have some opinions on how this test should be written. Below is my code:
```solidity
function testGetFee_ShouldCalculateCorrectValue(uint8 _burnFeePercentage, uint96 _amount) public {
   // We always think about fee as a percentage first.
   // We multiply by 100 later so that the code works
   if ((_burnFeePercentage * 100) > 1000) {
			return;
		}
		orchestrator.setBurnFee(ethVault, _burnFee * 100);
     // By dividing by 100 in the formula below, we can ensure that fee calculated 
     // is the same as the previous version
		uint256 calculatedFee = (ethVault.TCAPPrice() * (_amount) * (_burnFeePercentage)) / (100) / (ethVault.getOraclePrice(ethOracle));

		//execution
		uint256 currentFee = ethVault.getFee(_amount);

		// We assert that the old fee calculation is the same as the new one. 
		assertEq(calculatedFee, currentFee);
}
```

Reply:
This might bring some confusion as we do save the fee value in numbers between 1 and 1000, Having a multiplication like that just in the tests can confuse us somehow, I would suggest just adding that to the docs.